packages/rocketchat-lib/server/functions/setEmail.js

```diff
@@ -32,5 +32,5 @@ RocketChat._setEmail = function(userId, email) {
 };
 
 RocketChat.setEmail = RocketChat.RateLimiter.limitFunction(RocketChat._setEmail, 1, 60000, {
-	0: function(userId) { return !RocketChat.authz.hasPermission(userId, 'edit-other-user-info'); } // Administrators have permission to change others emails, so don't limit those
+	0: function(userId) { return !Meteor.userId() || !RocketChat.authz.hasPermission(Meteor.userId(), 'edit-other-user-info'); } // Administrators have permission to change others emails, so don't limit those
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['userId' is defined but never used.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2334233417/issues/source?bid=3199634&fileBranchId=3208512#l35)